genapi/genapi.go

```diff
@@ -692,11 +706,36 @@ func (g *GenAPI) doLever() {
 	g.Lever.Parse()
 }
 
+func SRVClientPreprocess(m *dns.Msg, dc string) {
+	dc = hashDC(dc)
+	if dc == "" {
+		return
+	}
+	for i := range m.Answer {
+		if ansSRV, ok := m.Answer[i].(*dns.SRV); ok {
+			tar := ansSRV.Target
+			if dc == tar[:len(dc)] {
+				ansSRV.Priority = uint16(4)
+			}
+		}
+	}
+}
+
+func hashDC(dc string) string {
```

Comment:
might as well make this a method on genapi like `getDCHash`, and do the `ParamStr` call in here too.